crates/sui-indexer/src/apis/read_api.rs

```diff
@@ -121,11 +121,18 @@ impl<S: IndexerStore> ReadApi<S> {
                     is_descending,
                 )
             }
-            // TODO(gegaowp): input objects are tricky to retrive from
-            // SuiTransactionResponse, instead we should store the BCS
-            // serialized transaction and retrive from there.
-            // This is now blocked by the endpoint on FN side.
-            Some(TransactionFilter::InputObject(_input_obj_id)) => Ok(vec![]),
+            Some(TransactionFilter::InputObject(input_obj_id)) => {
+                let input_obj_seq = self
+                    .state
+                    .get_input_object_sequence_by_digest(cursor_str, is_descending)?;
+                self.state.get_transaction_digest_page_by_input_object(
+                    input_obj_id.to_string(),
+                    None,
+                    input_obj_seq,
+                    limit,
```

Comment:
why not `limit + 1` as below?

Reply:
this should be limit + 1 indeed, nice catch, I will add tests for more input in table and with pagination logics in a separate PR.

Reply:
also moved the limit + 1 logic for "query all txns" to this read_api to be consistent